### PR TITLE
Add Algolia synonyms for searches with no results

### PR DESCRIPTION
**Pull request by rmloveland**
According to our internal dashboard called 'Algolia Metrics', we returned no search results for the following terms:

- optgen
- writetooolderror
- material view

This change adds synonyms for each of them to our search config. In particular the 'writetooolderror' is important to address, since it relates to a specific type of transaction retry error:

https://www.cockroachlabs.com/docs/v23.1/transaction-retry-error-reference

### CODE REVIEW

**Review by nickvigilante — APPROVED** [review]
LGTM! Once we merge it, we'll be able to test the index a bit better.

Not relevant for the scope of this PR, but I hope that one day, we can get to a point where we have testing indexes that generate alongside each Netlify build where we can observe how our results rank.

**Comment by rmloveland** [comment]
Nick Vigilante ***@***.***> writes:

> LGTM! Once we merge it, we'll be able to test the index a bit better.

Awesome, thanks!

> Not relevant for the scope of this PR, but I hope that one day, we can
> get to a point where we have testing indexes that generate alongside
> each Netlify build where we can observe how our results rank.

Interesting, I look forward to learning more about this!

**Pull request merged**
